contextutil/context.go

```diff
@@ -32,3 +36,26 @@ func Self(ctx context.Context) (*identity.IDConfig, error) {
 	}
 	return identity.GetIdentityConfig(tc)
 }
+
+// Tenant extracts the TenanConfig from the given context value
+func Tenant(ctx context.Context) (*configstore.TenantConfig, error) {
+	tc, ok := ctx.Value(self).(*configstore.TenantConfig)
+	if !ok {
+		return nil, ErrSelfNotFound
+	}
+	return tc, nil
+}
+
+// CentContext updates a context with tenant info using the configstore, must only be used for api handlers
+func CentContext(ctx context.Context, config configstore.Service) (context.Context, error) {
```

Comment:
I believe that the contextutil/context.go should only be a helper wrapper for our logic. On those lines, I'd like more to extract the `config.GetTenant(id)` to the invoker (handlers) and just use the `NewCentrifugeContext(ctx, tc)` 
It is a bit more overhead to every handler call but keeps context simpler, What do u think?